engine/render/src/render/render.cpp

```diff
@@ -684,6 +684,9 @@ namespace dmRender
             if (ro->m_SetStencilTest)
                 ApplyStencilTest(render_context, ro);
 
+            if (ro->m_SetFaceWinding)
+                dmGraphics::SetFaceWinding(context, ro->m_FaceWinding);
```

Comment:
Given the classic state problem of OpenGL, where a state from a previous draw call leaks into the next one, do we have a problem with this right now?

If kind of feels like we do right? E.g. with the blend factors.

Reply:
I have no idea if it's currently a problem or not, but it's likely given a scenario where a component wants to set the face winding (or blend factors or something else) but only has one RO. There's currently no way of resetting that state after the dispatch so in that case the component has to issue an empty RO to clean the state to the previous state, but then how would the component know what the previous state was.. In old immediate mode opengl you can push/pop states to a stack, but that's not possible with modern gl.

I dunno, maybe we should remove all the "set<...>" functions and keep that state in the render module and reflect changes based on what the RO looks like so you'd have to be explicit with the state for every render object. Almost like a pipeline state vulkan.